app/src/main/java/org/dhis2/utils/customviews/PeriodAdapter.kt

```diff
@@ -70,5 +79,38 @@ private class PeriodAdapter(
         }
     }
 
-    override fun getItemCount() = datePeriods.size + 1
+    fun setInputPeriod(){
+        var isAllowed = false
+        for (inputPeriodModel in inputPeriods) {
+            do {
+                if ((currentDate.after(inputPeriodModel.initialPeriodDate()) || currentDate == inputPeriodModel.initialPeriodDate()) && currentDate.before(
+                        inputPeriodModel.endPeriodDate()
+                    )
+                    && (inputPeriodModel.openingDate() == null || inputPeriodModel.openingDate() != null && DateUtils.getInstance().today.after(
+                        inputPeriodModel.openingDate()
+                    )
+                            || DateUtils.getInstance().today == inputPeriodModel.openingDate())
+                    && (inputPeriodModel.closingDate() == null || inputPeriodModel.closingDate() != null && DateUtils.getInstance().today.before(
+                        inputPeriodModel.closingDate()
+                    ))
```

Comment:
I think readability improves a bit if you do this :  

fun isTodayBetweenInitialAndEndPeriod() : Boolean{
return ...
}
fun isTodayBetweenOpenAndCloseDate() :  Boolean{
return ...
}

And then

if (isTodayBetweenInitialAndEndPeriod() &&  isTodayBetweenOpenAndCloseDate()) {
                    isAllowed = true
                    datePeriods.add(currentDate)
                    lastDate = currentDate
}

Reply:
You're rigth!